### PR TITLE
feat: add append & delete functions

### PR DESCRIPTION
**Pull request by santiagocardo**
Add `append` & `delete` functions to standard expression callbacks.

### CODE REVIEW

File: lib/expression/callbacks/standard.ex [hunk 1]
```diff
@@ -1434,4 +1434,32 @@ defmodule Expression.Callbacks.Standard do
 
     rem(integer1, integer2)
   end
+
+  @doc """
+  Appends items from one list to another list.
+  """
+  @expression_doc expression: "append([\"A\", \"B\"], [\"C\", \"B\"])",
+                  result: ["A", "B", "C", "B"]
+  def append(ctx, first_list, second_list) do
```
Comment (fedme) [inline lib/expression/callbacks/standard.ex]: Usually a function called `append` will add one element to a list. If this takes 2 lists should we call it `concat` rather?

File: lib/expression/callbacks/standard.ex [hunk 2]
```diff
@@ -1434,4 +1434,32 @@ defmodule Expression.Callbacks.Standard do
...
+
+    Enum.concat(first_list, second_list)
+  end
+
+  @doc """
+  Deletes an element from a map by the given key.
+  """
+  @expression_doc expression: "delete(contact, \"gender\")",
+                  context: %{
+                    "contact" => %{
```
Comment (fedme) [inline lib/expression/callbacks/standard.ex]: Can we use a different example? Using `contact` here seems to suggest the you can remove fields forma contact profile with this function

File: test/expression_test.exs [hunk 3]
```diff
@@ -49,6 +49,14 @@ defmodule ExpressionTest do
       assert nil == Expression.evaluate!("@foo[100]", %{"foo" => ["baz", "bar"]})
     end
 
+    test "append lists" do
```
Comment (fedme) [inline test/expression_test.exs]: Can we add corner cases to the test suite? Like empty lists, nils, etc.

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: yeah, that's true. But there is also a function called `concatenate` that concatenates strings 😅

File: lib/expression/callbacks/standard.ex (hunk 2 above)
Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: good point!

File: test/expression_test.exs (hunk 3 above)
Comment (santiagocardo) [reply test/expression_test.exs]: I can do it but in the current **expression** implementation, it will actually throw an error.

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (fedme) [reply lib/expression/callbacks/standard.ex]: I see, very unfortunate! :D
Can we maybe have this append function accept both a list or a single element as the second argument?

Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: what about `merge`? maybe we can add support for maps too.

Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: or maybe just use `concat` as you suggested but also as we can see from [TextIt expressions](https://textit.in/mr/docs/expressions.html#function:concat).

File: test/expression_test.exs (hunk 3 above)
Comment (smn) [reply test/expression_test.exs]: I think https://github.com/turnhub/expression/pull/151 fixes this issue -- would appreciate a review there :)

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (smn) [reply lib/expression/callbacks/standard.ex]: either we're going to have to be very disciplined about naming conventions (follow Redis' command examples) or we need to start namespacing into modules.

File: test/expression_test.exs (hunk 3 above)
Comment (santiagocardo) [reply test/expression_test.exs]: Approved @smn!
But in the current implementation, [we are ok throwing errors](https://github.com/turnhub/expression/blob/develop/test/expression_test.exs#L332) when manipulating incompatible types such as `nil`, `map`, and `string`. That is something we can revisit later in another PR or lib version.

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: what do you think it's better?
- `append` supporting both, a single element and a list of elements
- `merge` supporting both, lists and maps
- `concat` just for lists

I need to wrap up this because some people are waiting for this right now.

Comment (fedme) [reply lib/expression/callbacks/standard.ex]: if `concatenate` already exists in the expressions we support, should we maybe make it work for lists too in addition to strings?

Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: yes, that's also a good option. I like it!

Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: @smn what do you think about this? ⬆️

File: test/expression_test.exs (hunk 3 above)
Comment (smn) [reply test/expression_test.exs]: yeah I think we're ok with that. We need to form some good opinions on how to handle errors in expressions anyway, both when they're used as a standalone block `foo > bar` and when they're using in a string `Hello @if(foo > bar, "Anonymous", "Member")`

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (smn) [reply lib/expression/callbacks/standard.ex]: I like that idea but keep in mind that concatenate() as it's defined in the spec takes any amount of string arguments and joins them. We'd need to then do the same for lists, so it would be like a combination of Enum.concat and Enum.reduce

**Comment by smn** [comment]
@santiagocardo happy for you to wrap this up as best as you see fit!

**Review by smn — APPROVED** [review]
I think @fedme 's review on this is great! Let's address those things as best we can and then merge.

File: lib/expression/callbacks/standard.ex (hunk 1 above)
Comment (santiagocardo) [reply lib/expression/callbacks/standard.ex]: Ok, I ended up using `append()` which is what two clients were asking for but also to simplify things 👍🏼

File: test/expression_test.exs (hunk 3 above)
Comment (santiagocardo) [reply test/expression_test.exs]: Agree @smn. We would likely change that behaviour in the next `expression` version.
